<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@fgv/ts-utils](./ts-utils.md) &gt; [failsWithDetail](./ts-utils.failswithdetail.md)

## failsWithDetail() function

Returns [DetailedFailure&lt;T, TD&gt;](./ts-utils.detailedfailure.md) with a supplied error message and detail.

**Signature:**

```typescript
export declare function failsWithDetail<T, TD>(message: string, detail?: TD): DetailedFailure<T, TD>;
```

## Parameters

<table><thead><tr><th>

Parameter


</th><th>

Type


</th><th>

Description


</th></tr></thead>
<tbody><tr><td>

message


</td><td>

string


</td><td>

The error message to be returned.


</td></tr>
<tr><td>

detail


</td><td>

TD


</td><td>

_(Optional)_ The event detail to be returned.


</td></tr>
</tbody></table>

**Returns:**

[DetailedFailure](./ts-utils.detailedfailure.md)<!-- -->&lt;T, TD&gt;

An [DetailedFailure&lt;T, TD&gt;](./ts-utils.detailedfailure.md) with the supplied error message and detail.

## Remarks

The `failsWithDetail` alias was added in release 5.0 for naming consistency with [fails](./ts-utils.fails.md)<!-- -->, which was added to avoid conflicts with test frameworks and libraries.

